src/js/angular/ttyg/controllers/agent-settings-modal.controller.js

```diff
@@ -132,42 +155,24 @@ function AgentSettingsModalController($scope, $uibModalInstance, SimilarityServi
         const message = decodeHTML(
             $translate.instant(
                 'ttyg.agent.create_agent_modal.form.fts_search.fts_disabled_message',
-                {repositoryEditPage: '#/repository/edit/' + $scope.activeRepositoryInfo.id}
+                {repositoryEditPage: '#/repository/edit/' + $scope.agentFormModel.repositoryId}
             )
         );
         return $sce.trustAsHtml(message);
     };
 
     /**
-     * Resolves the hint for the missing similarity index message. This is needed because the hint contains a html link
-     * that should be properly rendered.
-     * @return {*}
+     * Opens the 'Create Similarity' view in a new tab.
      */
-    $scope.getNoSimilarityIndexHelpMessage = () => {
-        // The hint contains a html link which should be properly rendered.
-        const message = decodeHTML(
-            $translate.instant(
-                'ttyg.agent.create_agent_modal.form.similarity_index.no_similarity_index_message',
-                {similarityIndexPage: '#/similarity'}
-            )
-        );
-        return $sce.trustAsHtml(message);
+    $scope.goToCreateSimilarityView = () => {
+        TTYGContextService.emit(TTYGEventName.GO_TO_CREATE_SIMILARITY_VIEW, {repositoryId: $scope.agentFormModel.repositoryId});
     };
 
     /**
-     * Resolves the hint for the missing retrieval connector message. This is needed because the hint contains a html
-     * link that should be properly rendered.
-     * @return {*}
+     * Opens the 'Connectors' view in a new tab.
      */
-    $scope.getNoRetrievalConnectorHelpMessage = () => {
-        // The hint contains a html link which should be properly rendered.
-        const message = decodeHTML(
-            $translate.instant(
-                'ttyg.agent.create_agent_modal.form.retrieval_search.no_retrieval_connectors_message',
-                {retrievalConnectorPage: '#/connectors'}
-            )
-        );
-        return $sce.trustAsHtml(message);
+    $scope.goToConnectorsView = () => {
+        TTYGContextService.emit(TTYGEventName.GO_TO_CONNECTORS_VIEW, {repositoryId: $scope.agentFormModel.repositoryId});
```

Comment:
but why?? Didn't we discussed that this controller is more like a smart component. You shoud've just did these confirmations here!